### PR TITLE
update client-support-angular to angular 17

### PR DESCRIPTION
**Pull request by VincentUllal**
**Description**

breaking changes!

**Related issue(s)**

#3659

### CODE REVIEW

**Comment by JohannesDoberer** [comment]
Cannot released in Luigi v2.x because it will not run in an angularv14 client.

File: client-frameworks-support/client-support-angular/package.json [hunk 1]
```diff
@@ -13,39 +13,39 @@
...
-    "@angular/platform-browser": "^14.2.12",
-    "@angular/platform-browser-dynamic": "^14.2.12",
-    "@angular/router": "^14.2.12",
-    "@luigi-project/client": "^2.0.0",
-    "@luigi-project/testing-utilities": "^2.0.0",
-    "rxjs": "~6.6.0",
+    "@angular/animations": "^17.0.0",
+    "@angular/common": "^17.0.0",
+    "@angular/compiler": "^17.0.0",
+    "@angular/core": "^17.0.0",
```
Comment (walmazacn) [inline client-frameworks-support/client-support-angular/package.json:19]: Btw, Angular 18 is out - but if we want to stick to v17 then at least bump it to v17.3.10

Comment (walmazacn) [reply client-frameworks-support/client-support-angular/package.json:19]: @VincentUllal Here's proposed list of up-to-date dependencies which works for me - bear in mind you might need to remove both 'node_modules' and 'package-lock.json' locally to install everything with `npm install` command:
```
"dependencies": {
  "@angular/animations": "^17.3.11",
  "@angular/common": "^17.3.11",
  "@angular/compiler": "^17.3.11",
  "@angular/core": "^17.3.11",
  "@angular/forms": "^17.3.11",
  "@angular/platform-browser": "^17.3.11",
  "@angular/platform-browser-dynamic": "^17.3.11",
  "@angular/router": "^17.3.11",
  "@luigi-project/client": "^2.13.0",
  "@luigi-project/testing-utilities": "^2.13.0",
  "rxjs": "^7.4.0",
  "tslib": "^2.4.0",
  "zone.js": "~0.14.0"
},
"devDependencies": {
  "@angular-devkit/build-angular": "^17.3.8",
  "@angular/cli": "^17.3.8",
  "@angular/compiler-cli": "^17.3.11",
  "@types/jasmine": "~5.1.0",
  "@types/node": "^18.11.9",
  "codelyzer": "^6.0.0",
  "jasmine-core": "~5.1.0",
  "jasmine-spec-reporter": "~5.0.0",
  "karma": "~6.4.0",
  "karma-chrome-launcher": "~3.2.0",
  "karma-coverage": "~2.2.0",
  "karma-jasmine": "~5.1.0",
  "karma-jasmine-html-reporter": "~2.1.0",
  "ng-packagr": "^17.3.0",
  "protractor": "~7.0.0",
  "ts-node": "~8.3.0",
  "tslint": "~6.1.0",
  "typescript": "~5.2.2"
}
```

Comment (VincentUllal) [reply client-frameworks-support/client-support-angular/package.json]: Works for me as well. Thanks for the help :)

**Review by walmazacn — APPROVED** [review]
LGTM 👍

**Pull request merged**